pkg/notify/elasticsearch.go

```diff
@@ -0,0 +1,100 @@
+package notify
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/infracloudio/botkube/pkg/config"
+	"github.com/infracloudio/botkube/pkg/events"
+	log "github.com/infracloudio/botkube/pkg/logging"
+	"github.com/olivere/elastic"
+)
+
+// ElasticSearch contains auth cred and index setting
+type ElasticSearch struct {
+	Username string
+	Password string
+	Server   string
+	Index    string
+	Shards   int
+	Replicas int
+	Type     string
+}
+
+// NewElasticSearch returns new Slack object
+func NewElasticSearch(c *config.Config) Notifier {
+	return &ElasticSearch{
+		Username: c.Communications.ElasticSearch.Username,
+		Password: c.Communications.ElasticSearch.Password,
+		Server:   c.Communications.ElasticSearch.Server,
+		Index:    c.Communications.ElasticSearch.Index.Name,
+		Type:     c.Communications.ElasticSearch.Index.Type,
+		Shards:   c.Communications.ElasticSearch.Index.Shards,
+		Replicas: c.Communications.ElasticSearch.Index.Replicas,
+	}
+}
+
+type mapping struct {
+	Settings settings `json:"settings"`
+}
+
+type settings struct {
+	Index index `json:"index"`
+}
+type index struct {
+	Shards   int `json:"number_of_shards"`
+	Replicas int `json:"number_of_replicas"`
+}
+
+// SendEvent sends event notification to slack
+func (e *ElasticSearch) SendEvent(event events.Event) error {
+	log.Logger.Debug(fmt.Sprintf(">> Sending to ElasticSearch: %+v", event))
+	ctx := context.Background()
+	client, err := elastic.NewClient(elastic.SetURL(e.Server), elastic.SetBasicAuth(e.Username, e.Password), elastic.SetSniff(false), elastic.SetHealthcheck(false), elastic.SetGzip(true))
```

Comment:
are we doing auth for sending every event ?

Reply:
What all options are available apart from basic auth ?

Reply:
We should initialize client during init() and reuse it. ElasticSearch doesn't have any other authentication method. The filebeat also uses basic auth for authentication
https://www.elastic.co/guide/en/beats/filebeat/current/elasticsearch-output.html

Reply:
Done!